docs/analytics_tools/github_setup.md

```diff
@@ -3,46 +3,47 @@
-    * Run the following command to "start" ssh key in the background, which should return a response similar to `Agent pid 258`
-        * `eval "$(ssh-agent -s)"`
-    * Add the ssh key to the ssh agent with this command
-        * `ssh-add ~/.ssh/id_ed25519`
-    * From here we will copy the contents of your public ssh file, which we will first do by viewing it's contents.
-        * To view:
-            * `cat ~/.ssh/id_ed25519.pub`
-        * Then, select and copy the entire contents of the file that display
-            * The text should begin with `ssh-ed25519` and end with your GitHub email address
-            * Beware of copying white spaces, which may cause errors in the following steps
-
-
-    * Once copied, navigate to GitHub
-    * In the top right corner, select `Settings` and then select `SSH and GPC Keys` in the left sidebar
-    * Select `New SSH Key`
-        * For the title add something like ‘JupyterHub’
-        * Paste the key that you copied into key field
-    * Click `Add SSH Key`, and, if necessary, enter password
-
-
-    * After you've added your SSH key to your GitHub account, open a new Jupyter terminal window and you'll be able to test your connection with:
-        * `ssh -T git@github.com`
+2. Open a terminal in JupyterHub. All of our commands will be typed in this terminal.
+3. `gh auth login` and select the following options
+```
+(base) jovyan@f4b18b106c18:~$ gh auth login
+? What account do you want to log into? GitHub.com
+? What is your preferred protocol for Git operations? SSH
+? Generate a new SSH key to add to your GitHub account? Yes
+? Enter a passphrase for your new SSH key (Optional)
+? Title for your SSH key: GitHub CLI
+? How would you like to authenticate GitHub CLI? Login with a web browser
+```
+You can press `Enter` to leave the passphrase empty, or you may provide a password; in the future, you will need to enter this password when your server starts. If you've already created an SSH key, you will be prompted to select the existing key rather than creating a new one.
+
+4. You will then be given a one-time code and instructed to press `Enter` to open a web browser, which will fail if you are using JupyterHub. However, you can manually open the link in a browser and enter the code. You will end up with output similar to the following.
+```
+! First copy your one-time code: ABCD-1234
+Press Enter to open github.com in your browser...
+...
+! Failed opening a web browser at https://github.com/login/device
+  exit status 3
+  Please try entering the URL in your browser manually
+✓ Authentication complete.
+- gh config set -h github.com git_protocol ssh
+✓ Configured git protocol
+✓ Uploaded the SSH key to your GitHub account: /home/jovyan/.ssh/id_ed25519.pub
+✓ Logged in as atvaccaro
+```
 
 After completing the steps above be sure to complete the section below to persist your SSH key between sessions and enable extensions.
```

Comment:
Is this still required? (persist the SSH key etc.)

Reply:
Yes, `profile.sh` does configure the key with keychain but if a `.bashrc` file exists, it'll ignore `profile.sh` unless you specifically source it (as best I can tell)